backend/src/modules/jokes_modules/getRandomiseJokesFromDatabase.js

```diff
@@ -0,0 +1,35 @@
+import { getJokesCollection } from "../db_modules/dbClientConnect.js";
+
+export const getRandomiseJokesFromDatabase = async objOfParams => {
+  const jokesCollection = getJokesCollection();
+
+  let categoryQuery = setCategoryQuery(objOfParams.selectedCategory);
+
+  let searchInputTextQuery = setSearchInputTextQuery(
+    objOfParams.searchInputText
+  );
+
+  const numberOfJokesQuery = { size: parseInt(objOfParams.numberOfJokes) };
+
+  return await jokesCollection
+    .aggregate([
+      { $match: searchInputTextQuery },
+      { $match: categoryQuery },
+      { $sample: numberOfJokesQuery }
+    ])
+    .toArray();
+};
+
+const setCategoryQuery = selectedCategory => {
+  if (selectedCategory !== "all") {
+    return { categories: selectedCategory };
+  }
+  return {};
+};
+
+const setSearchInputTextQuery = searchInputText => {
+  if (searchInputText !== "empty_search_input") {
+    return { value: new RegExp(searchInputText, "gi") };
+  }
```

Comment:
I would reverse this. If searchInputText is equal to `empty_***`, then I would returned empty object, and after that if continued with the rest of logic. Return fast principle :)